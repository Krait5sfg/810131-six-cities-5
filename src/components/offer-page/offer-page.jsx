import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {getRating} from '../../utils/common';
import {TypeAccommodation} from '../../utils/const';
import ReviewForm from '../review-form/review-form';
import {OfferPropTypes, ReviewPropTypes} from '../../utils/property-type';
import ReviewList from '../review-list/review-list';
import Map from '../map/map';
import {TypePage} from '../../utils/const';
import PlaceCardList from '../place-card-list/place-card-list';
import {connect} from 'react-redux';
import User from '../user/user';
import {AuthorizationStatus} from '../../utils/const';
import {getActiveOffer} from '../../store/api-actions';


const MAX_IMAGE_ON_PAGE = 6;
const MAX_OFFER_ON_PAGE = 3;

class OfferPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {idActiveOffer, updateActiveOffer} = this.props;

    // Запрос на сервер для получения активного предложения
    updateActiveOffer(idActiveOffer);
  }

  render() {
    const {offer} = this.props;
    if (Object.keys(offer).length) {
      const {offers, reviews, onLinkEmailClick, city, authorizationStatus} = this.props;
      const {id, images, accommodation, host, description, isFavorite} = offer;
      const {isPremium, rating, title, type, bedroomsCount, guestsLimit, price, features} = accommodation;

      // три предложения за исключением выведенного на страницу
      const otherOffers = offers.filter((offering) => offering.id !== id && offering.city === city).slice(0, MAX_OFFER_ON_PAGE);

      const offerPageReviews = reviews
        .filter((review) => review.offerId === id)
        .sort((firstReview, secondReview) => firstReview < secondReview ? -1 : 1);


      const favoriteButtonClass = isFavorite ? `property__bookmark-button--active` : ``;

      const forRenderImages = images.length > MAX_IMAGE_ON_PAGE ? images.slice(0, MAX_IMAGE_ON_PAGE) : [...images];

      const imagesElements = forRenderImages.map((image, index) => {
        return (
          <div className="property__image-wrapper" key={index}>
            <img className="property__image" src={image} alt="Photo studio" />
          </div>
        );
      });

      const featuresElements = features.map((feature, index) => {
        return (
          <li className="property__inside-item" key={index}>
            {feature}
          </li>
        );
      });

      const premiumMarkElement = isPremium ? <div className="property__mark"><span>Premium</span></div> : ``;

      return (
        <div className="page">
          <header className="header">
            <div className="container">
              <div className="header__wrapper">
                <div className="header__left">
                  <a className="header__logo-link" href="/">
                    <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
                  </a>
                </div>
                <nav className="header__nav">
                  <ul className="header__nav-list">
                    <User onLinkEmailClick={onLinkEmailClick} />
                  </ul>
                </nav>
              </div>
            </div>
          </header>

          <main className="page__main page__main--property">
            <section className="property">
              <div className="property__gallery-container container">
                <div className="property__gallery">
                  {imagesElements}
                </div>
              </div>
              <div className="property__container container">
                <div className="property__wrapper">
                  {premiumMarkElement}
                  <div className="property__name-wrapper">
                    <h1 className="property__name">
                      {title}
                    </h1>
                    <button className={`property__bookmark-button button ${favoriteButtonClass}`} type="button">
                      <svg className="property__bookmark-icon" width="31" height="33">
                        <use xlinkHref="#icon-bookmark" />
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="property__rating rating">
                    <div className="property__stars rating__stars">
                      <span style={{width: getRating(rating)}}></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                    <span className="property__rating-value rating__value">{rating}</span>
                  </div>
                  <ul className="property__features">
                    <li className="property__feature property__feature--entire">
                      {TypeAccommodation[type.toUpperCase()]}
                    </li>
                    <li className="property__feature property__feature--bedrooms">
                      {bedroomsCount} Bedrooms
                    </li>
                    <li className="property__feature property__feature--adults">
                      Max {guestsLimit} adults
                    </li>
                  </ul>
                  <div className="property__price">
                    <b className="property__price-value">&euro;{price}</b>
                    <span className="property__price-text">&nbsp;{` `}night</span>
                  </div>
                  <div className="property__inside">
                    <h2 className="property__inside-title">What&apos;s inside</h2>
                    <ul className="property__inside-list">
                      {featuresElements}
                    </ul>
                  </div>
                  <div className="property__host">
                    <h2 className="property__host-title">Meet the host</h2>
                    <div className="property__host-user user">
                      <div className={`property__avatar-wrapper ${host.isPro ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                        <img className="property__avatar user__avatar" src={host.avatar} width="74" height="74" alt="Host avatar" />
                      </div>
                      <span className="property__user-name">
                        {host.name}
                      </span>
                    </div>
                    <div className="property__description">
                      <p className="property__text">
                        {description}
                      </p>
                    </div>
                  </div>
                  <section className="property__reviews reviews">
                    <ReviewList offerPageReviews={offerPageReviews} />
                    {authorizationStatus === AuthorizationStatus.AUTH ?
                      <ReviewForm /> : ``}
                  </section>
                </div>
              </div>
              <Map typePage={TypePage.OFFER} city={city} offers={otherOffers} />
            </section>
            <div className="container">
              <section className="near-places places">
                <h2 className="near-places__title">Other places in the neighbourhood</h2>
                <PlaceCardList offers={otherOffers} typePage={TypePage.OFFER} />
              </section>
            </div>
          </main>
        </div >
      );
    }
    return <h2 style={{position: `absolute`, top: `50%`, left: `50%`, transform: `translate(-50%, -50%)`}}>Loading...</h2>;
  }
}

OfferPage.propTypes = {
  onLinkEmailClick: PropTypes.func.isRequired,
  offer: PropTypes.oneOfType([
    PropTypes.object.isRequired,
    OfferPropTypes
  ]),
  offers: PropTypes.arrayOf(OfferPropTypes).isRequired,
  reviews: PropTypes.arrayOf(ReviewPropTypes).isRequired,
  city: PropTypes.string.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  idActiveOffer: PropTypes.number.isRequired,
  updateActiveOffer: PropTypes.func.isRequired
};

// связывает store c пропсами компонента
const mapStateToProps = (({PROCESS, USER, DATA}) => ({
  city: PROCESS.city,
  authorizationStatus: USER.authorizationStatus,
  idActiveOffer: PROCESS.idActiveOffer,
  offer: DATA.activeOffer
}));

const mapDispatchToProps = ((dispatch) => ({
  updateActiveOffer(id) {
    dispatch(getActiveOffer(id));
  }
}));

export {OfferPage};
export default connect(mapStateToProps, mapDispatchToProps)(OfferPage);
