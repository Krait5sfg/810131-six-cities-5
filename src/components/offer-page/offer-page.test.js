import React from 'react';
import renderer from 'react-test-renderer';
import {OfferPage} from './offer-page';

const noop = () => {};
const offer = {
  city: ``,
  accommodation: {
    bedroomsCount: 2,
    features: [`Laptop friendly workspace`, `Baby seat`, `Air conditioning`],
    guestsLimit: 2,
    isPremium: true,
    price: 200,
    rating: 4,
    title: `The Joshua Tree House`,
    type: `room`,
  },
  cityLocation: {latitude: 0, longitude: 0, zoom: 0},
  coordinates: [0, 0],
  description: `Discover daily local life in city center...`,
  host: {
    avatar: `img/avatar-angelina.jpg`,
    isPro: true,
    name: `Angelina`,
  },
  id: 1,
  images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`],
  isFavorite: false,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
};

const nearbyOffers = [
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: false,
      price: 300,
      rating: 4.5,
      title: `The Joshua Tree House`,
      type: `hotel`,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 2,
    images: [],
    isFavorite: false,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: false,
      price: 300,
      rating: 4.5,
      title: `The Joshua Tree House`,
      type: `hotel`,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 3,
    images: [],
    isFavorite: false,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: false,
      price: 300,
      rating: 4.5,
      title: `The Joshua Tree House`,
      type: `hotel`,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 4,
    images: [],
    isFavorite: false,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
  },
];

jest.mock(`../header/header`, () => `Header`);
jest.mock(`../user/user`, () => `User`);
jest.mock(`../loading/loading`, () => `Loading`);
jest.mock(`../place-card-list/place-card-list`, () => `PlaceCardList`);
jest.mock(`../map/map`, () => `Map`);
jest.mock(`../review-form/review-form`, () => `ReviewForm`);
jest.mock(`../review-list/review-list`, () => `ReviewList`);

describe(`OfferPage render correctly`, () => {
  it(`OfferPage with authorization`, () => {
    const tree = renderer
      .create(<OfferPage
        onLinkEmailClick={noop}
        offer={offer}
        nearbyOffers={nearbyOffers}
        authorizationStatus={`AUTH`}
        idActiveOffer={1}
        updateActiveOffer={noop}
        updateNearbyOffers={noop}
        onFavoriteButtonClick={noop}
        updateFavoriteStatus={noop}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`OfferPage without authorization`, () => {
    const tree = renderer
      .create(<OfferPage
        onLinkEmailClick={noop}
        offer={offer}
        nearbyOffers={nearbyOffers}
        authorizationStatus={`NO_AUTH`}
        idActiveOffer={1}
        updateActiveOffer={noop}
        updateNearbyOffers={noop}
        onFavoriteButtonClick={noop}
        updateFavoriteStatus={noop}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
