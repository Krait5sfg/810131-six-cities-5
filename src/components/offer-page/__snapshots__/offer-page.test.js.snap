// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPage render correctly OfferPage with authorization 1`] = `
<div
  className="page"
>
  <Header>
    <User
      onLinkEmailClick={[Function]}
    />
  </Header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The Joshua Tree House
            </h1>
            <button
              className="property__bookmark-button button "
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Private Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              2
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              200
            </b>
            <span
              className="property__price-text"
            >
               
               
              night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Discover daily local life in city center...
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <ReviewList
              idActiveOffer={1}
            />
            <ReviewForm
              idActiveOffer={1}
            />
          </section>
        </div>
      </div>
      <Map
        city=""
        typePage="offer"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <PlaceCardList
          offers={
            Array [
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 2,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 3,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 4,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
            ]
          }
          onFavoriteButtonClick={[Function]}
          typePage="offer"
        />
      </section>
    </div>
  </main>
</div>
`;

exports[`OfferPage render correctly OfferPage without authorization 1`] = `
<div
  className="page"
>
  <Header>
    <User
      onLinkEmailClick={[Function]}
    />
  </Header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The Joshua Tree House
            </h1>
            <button
              className="property__bookmark-button button "
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Private Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              2
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              200
            </b>
            <span
              className="property__price-text"
            >
               
               
              night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Discover daily local life in city center...
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <ReviewList
              idActiveOffer={1}
            />
            
          </section>
        </div>
      </div>
      <Map
        city=""
        typePage="offer"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <PlaceCardList
          offers={
            Array [
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 2,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 3,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
              Object {
                "accommodation": Object {
                  "bedroomsCount": 0,
                  "features": Array [],
                  "guestsLimit": 4,
                  "isPremium": false,
                  "price": 300,
                  "rating": 4.5,
                  "title": "The Joshua Tree House",
                  "type": "hotel",
                },
                "city": "",
                "cityLocation": Object {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0,
                },
                "coordinates": Array [],
                "description": "",
                "host": Object {
                  "avatar": "",
                  "isPro": true,
                  "name": "",
                },
                "id": 4,
                "images": Array [],
                "isFavorite": false,
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              },
            ]
          }
          onFavoriteButtonClick={[Function]}
          typePage="offer"
        />
      </section>
    </div>
  </main>
</div>
`;
